From 0f91836a5eb9e71485453ffafc281079a4ebf01d Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 18 Jul 2018 09:46:53 +0300
Subject: [PATCH] PCIE: preboot is first to be called from BTL fixup

Several default registers values of the PCIe PHY were found incorrect.
Adjust those registers early enough to correct the PCIe PHY configuration.

SW-007

Change-Id: Iba4ea576d23aae96a66fb81a116a4d2d1812a783
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot.c b/preboot.c
index d99f49e..b5cf548 100644
--- a/preboot.c
+++ b/preboot.c
@@ -105,6 +105,18 @@
 	/* Errata: disable unused design that cause PCIe phy SRAM DERR */
 	writel(0, mmPCIE_PHY_MEM_WR_MC);
 
+	/* Adjust the incorrect PHY protocol0 defaults */
+	writel(0xA067, mmPCIE_PHY_PROTOCOL0_EXT_MPLLA_BANDWIDTH);
+	writel(0x19, mmPCIE_PHY_PROTOCOL0_EXT_MPLLA_MULTIPLIER);
+	writel(0xA01E, mmPCIE_PHY_PROTOCOL0_EXT_MPLLB_BANDWIDTH);
+	writel(0x50, mmPCIE_PHY_PROTOCOL0_EXT_MPLLB_MULTIPLIER);
+	writel(0xE, mmPCIE_PHY_PROTOCOL0_EXT_RX_MISC_G1_SLICE0);
+	writel(0xC, mmPCIE_PHY_PROTOCOL0_EXT_RX_MISC_G2_SLICE0);
+	writel(0xA, mmPCIE_PHY_PROTOCOL0_EXT_RX_MISC_G3_SLICE0);
+	writel(0x9, mmPCIE_PHY_PROTOCOL0_EXT_RX_MISC_G4_SLICE0);
+	writel(0x4, mmPCIE_PHY_PROTOCOL0_EXT_TX_VBOOST_LVL);
+	writel(0x1, mmPCIE_PHY_PHY_EXT_CTRL_SEL);
+
 	/* Clear phy_init (remove the workaround provided by SNPS)*/
 	writel(0, mmPCIE_AUX_PHY_INIT);
 
